#184581175 User Login feature

GasanaJr opened the pull request:
#### What does this PR do?
This PR adds a login functionality to the application
#### Description of Task to be completed?
- add a login endpoint
- Users should be able to login via email and password
- add unit testing for the login endpoint
#### How should this be manually tested?
- clone this repo `git clone https://github.com/atlp-rwanda/e-comm-team-amigos-bn`
- switch your working directory `cd e-comm-team-amigos-bn`
- create an `.env` and update it with `.env-example`
- install all dependencies by running `npm install`
- in your terminal run `npm start`
#### Test the endpoints using Postman:
- `POST user/login`
#### Unit testing
- in your terminal run `npm run test`
#### Any background context you want to provide?
- PostgreSQL installed in your local machine
- Node JS installed in your local machine
#### Screenshot
- <img width="961" alt="Screenshot 2023-03-13 at 3 23 28 PM" src="https://user-images.githubusercontent.com/43322242/224723204-d7d906c5-47cc-457a-8f85-b520d769c820.png">- <img width="1388" alt="Screenshot 2023-03-11 at 10 35 52 AM" src="https://user-images.githubusercontent.com/43322242/224476007-8f6dc662-1b81-4728-9b89-996c4d6379bd.png">- 
- #### What are the relevant pivotal tracker stories?
Story ID: [#184581175](https://www.pivotaltracker.com/n/projects/2632829/stories/184581175)

GasanaJr committed:
feat(user): User login and testing

- add user login endpoint
- tests the endpoint using mocha and chai

[Finishes #184581175]

ericndungutse reviewed (approved):
LGTM

nshutijonathan reviewed (changes requested):
You have 7 commits messages, can you please narrow them down to 1?

GasanaJr commented:
> You have 7 commits messages, can you please narrow them down to 1?
Of the seven commits on this branch, only two belong to me. The other commits came with me rebasing from the Signup branch. I am not sure if squashing them won't lead to any data loss.

nyakamwe committed:
ft(signUp):sign up a user
- creates signUp endpoint
- validate user inputs
- validate user inputs if are valid
- generate jwt on successfull user registration
- create a unut testing for signUp
- documents user signUp endpoint with swager ui

[Delivers #184581172]

feat(user): User signup

- add user endpoint
- add user validation to ensure that user enters valid email, password and username
- rename index file to user in a folder called controllers

[Delivers #184581172]

feat(user): User signup

- add user endpoint
- add user validation to ensure that user enters valid email, password and username
- rename index file to user in a folder called controllers

[Delivers #184581172]

ft(signUp):sign up a user

- creates signUp endpoint
- validate user inputs
- generate jwt on successfull user registration
- creates a middleware function for verifying all protected routes
- documents both user signUp and login

[Delivers #184581172]

ft(login): User Login

- create user login endpoint

[Finishes #184581175]
